accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/impl/AcConfigChangeTracker.java

```diff
@@ -41,7 +41,7 @@ public boolean configIsUnchangedComparedToLastExecution(Map<String, String> conf
                 }
                 
                 acHistoryNode.setProperty(executionKey, hashOfConfigFilesThisExecution);
-                // save upon regular session save at the very end
+                session.save(); // save directly to avoid parallel executions of the exact same config in AEMaaCS 
```

Comment:
Although this will reduce the probability of the race condition this won't prevent it. It is still possible that two pods almost immediately execute the same config. I think we need rather a marker at the beginning of the execution indicating that some pod is starting with processing the config....

Reply:
yes, this just improves the situation (but by a fair bit). 

What I would like to avoid is the block parallel executions entirely, so the additional new marker (the current marker cannot be used as it relies already the config file content hash) would have to be set in the beginning and removed after the above marker is set (and for intermediate exception cases, it has be removed in a finally block somewhere).